core/server/src/main/java/alluxio/master/file/meta/InodeTree.java

```diff
@@ -585,7 +585,7 @@ public CreatePathResult createPath(LockedInodePath inodePath, CreatePathOptions<
     Inode<?> lastInode = currentInodeDirectory.getChild(name);
     if (lastInode != null) {
       // Lock the last inode before subsequent operations, and add it to the lock group.
-      lockList.lockWrite(lastInode);
+      lockList.lockWriteAndCheckFullPath(lastInode, currentInodeDirectory, name);
```

Comment:
do you have a guideline stating when we should lock and check parent, and when we should lock and check full path (or, parent + name) to replace original lock ?